qa/purchase_offline_error.py

```diff
@@ -0,0 +1,129 @@
+import requests
+import json
+import time
+from collections import OrderedDict
+from test_framework.test_framework import OpenBazaarTestFramework, TestFailure
+
+
+class PurchaseOfflineErrorTest(OpenBazaarTestFramework):
+
+    def __init__(self):
+        super().__init__()
+        self.num_nodes = 3
+
+    def run_test(self):
+        alice = self.nodes[0]
+        bob = self.nodes[1]
+
+        # post profile for alice
+        with open('testdata/profile.json') as profile_file:
+            profile_json = json.load(profile_file, object_pairs_hook=OrderedDict)
+        api_url = alice["gateway_url"] + "ob/profile"
+        requests.post(api_url, data=json.dumps(profile_json, indent=4))
+
+        # post listing to alice
+        with open('testdata/listing.json') as listing_file:
+            listing_json = json.load(listing_file, object_pairs_hook=OrderedDict)
+        if self.bitcoincash:
+            listing_json["metadata"]["pricingCurrency"] = "tbch"
+
+        api_url = alice["gateway_url"] + "ob/listing"
+        r = requests.post(api_url, data=json.dumps(listing_json, indent=4))
+        if r.status_code == 404:
+            raise TestFailure("PurchaseOfflineErrorTest - FAIL: Listing post endpoint not found")
+        elif r.status_code != 200:
+            resp = json.loads(r.text)
+            raise TestFailure("PurchaseOfflineErrorTest - FAIL: Listing POST failed. Reason: %s", resp["reason"])
+        time.sleep(20)
+
+        # get listing hash
+        api_url = alice["gateway_url"] + "ipns/" + alice["peerId"] + "/listings.json"
+        r = requests.get(api_url)
+        if r.status_code != 200:
+            raise TestFailure("PurchaseOfflineErrorTest - FAIL: Couldn't get listing index")
+        resp = json.loads(r.text)
+        listingId = resp[0]["hash"]
+
+        # bob fetch listing to cache
+        api_url = bob["gateway_url"] + "ipfs/" + listingId
+        requests.get(api_url)
+
+        # generate some coins and send them to bob
+        api_url = bob["gateway_url"] + "wallet/address"
+        r = requests.get(api_url)
+        if r.status_code == 200:
+            resp = json.loads(r.text)
+            address = resp["address"]
+        elif r.status_code == 404:
+            raise TestFailure("PurchaseOfflineErrorTest - FAIL: Address endpoint not found")
+        else:
+            raise TestFailure("PurchaseOfflineErrorTest - FAIL: Unknown response")
+        self.send_bitcoin_cmd("sendtoaddress", address, 10)
+        time.sleep(3)
+
+        # shutdown alice
+        api_url = alice["gateway_url"] + "ob/shutdown"
+        requests.post(api_url, data="")
+        time.sleep(4)
+
+        # bob send order
+        with open('testdata/order_direct.json') as order_file:
+            order_json = json.load(order_file, object_pairs_hook=OrderedDict)
+        order_json["items"][0]["listingHash"] = listingId
+
+        # set empty shipping address to trigger error
```

Comment:
If an empty shipping address would produce a PROCESSING_ERROR, shouldn't the purchase API be rejecting that purchase with an error in the first place?